### PR TITLE
Implement thread parking for xous

### PR DESCRIPTION
**Pull request by joboet**
This follows the pattern set by [the Windows parker](https://github.com/rust-lang/rust/blob/ddef56d5dfa18f169af9db912dc8e8343797eebb/library/std/src/sys/windows/thread_parking.rs) when it uses keyed events. An atomic variable is used to track the state and optimize the fast path, while notifications are send via the ticktime server to block and unblock the thread.

ping @xobs
@rustbot label +T-libs +A-atomic
r? libs

### CODE REVIEW

File: library/std/src/sys/xous/thread_parking.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        }
+    }
+
+    pub fn unpark(self: Pin<&Self>) {
+        let state = self.state.swap(NOTIFIED, Release);
+        if state == PARKED {
+            // The thread is parked, wake it up.
+            blocking_scalar(
+                ticktimer_server(),
+                crate::os::xous::services::TicktimerScalar::NotifyCondition(self.index(), 1).into(),
```
Comment (xobs) [inline library/std/src/sys/xous/thread_parking.rs:83]: You might try importing `TicktimerScalar` at the top here to avoid qualifying the entire thing

**Review by xobs — COMMENTED** [review]
Just need to make the one change, and I think it's good to merge.

It's very similar to a local patch I'm carrying while trying to debug threading issues. I have this particular patch disabled in my local branch in order to eliminate potential issues with condvars. Thread parking improves features such as channels, so I think this is good to merge.

File: library/std/src/sys/xous/thread_parking.rs (hunk 1 above)
Comment (joboet) [reply library/std/src/sys/xous/thread_parking.rs]: Oh yes, I forgot about this one.

**Comment by joboet** [comment]
Sorry, wrong branch 🤦

**Pull request merged**